### PR TITLE
Cell feature

### PR DESCRIPTION
**Pull request by Arique1104**

### CODE REVIEW

File: lib/cell.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
+class Cell
+ attr_reader  :coordinate,
+              :ship
+
+  def initialize(coordinate)
+    @coordinate = coordinate
+    @ship = nil
+    @empty = true
+    @fired_upon = false
```
Comment (priyapower) [inline lib/cell.rb:9]: The fired_upon was stumping me! After seeing your code, I was like   ;__;  I couldn't believe how silly I felt. :)

**Comment by priyapower** [comment]
Question: If I want to pull this file to test, I would switch to the branch it's on and do git pull or git fetch? This is what I'm still confused about?

**Comment by priyapower** [comment]
Never mind, found what I was looking for [here](https://www.freecodecamp.org/news/git-fetch-vs-pull/)

File: test/cell_test.rb [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+    assert_equal "X", cell_2.render
+    assert_equal true, cruiser.sunk?
+
+
+  end
+
+
+
+end
```
Comment (priyapower) [inline test/cell_test.rb:80]: First Pry Session (Testing):
[1] pry(main)> require './lib/ship'
=> true
[2] pry(main)> require './lib/cell'
=> true
[3] pry(main)> cell = Cell.new("B4")
=> #<Cell:0x00007fd84282f790
 @coordinate="B4",
 @empty=true,
 @fired_upon=false,
 @ship=nil>
[4] pry(main)> cruiser = Ship.new("Cruiser", 3)
=> #<Ship:0x00007fd840b5ae80 @health=3, @length=3, @name="Cruiser">
[5] pry(main)> cell.place_ship(cruiser)
=> #<Ship:0x00007fd840b5ae80 @health=3, @length=3, @name="Cruiser">
[6] pry(main)> cell.fired_upon?
=> false
[7] pry(main)> cell.fire_upon
=> 2
[8] pry(main)> cell.ship_health
NoMethodError: undefined method `ship_health' for #<Cell:0x00007fd84282f790>
from (pry):8:in `__pry__'
[9] pry(main)> cell.ship.health
=> 2
[10] pry(main)> cell.fired_upon?
=> true

Comment (priyapower) [reply test/cell_test.rb:80]: Second Pry Session (Testing):
[1] pry(main)> require './lib/ship'
=> true
[2] pry(main)> require './lib/cell'
=> true
[3] pry(main)> cell_1 = Cell.new("B4")
=> #<Cell:0x00007ff7f117f188
 @coordinate="B4",
 @empty=true,
 @fired_upon=false,
 @ship=nil>
[4] pry(main)> cell_1.render
=> "."
[5] pry(main)> cell_1.fire_upon
=> nil
[6] pry(main)> cell_1.render
=> "M"
[7] pry(main)> cell_2 = Cell.new("C3")
=> #<Cell:0x00007ff7f1176600
 @coordinate="C3",
 @empty=true,
 @fired_upon=false,
 @ship=nil>
[8] pry(main)> cruiser = Ship.new("Cruiser", 3)
=> #<Ship:0x00007ff7f2903ac0 @health=3, @length=3, @name="Cruiser">
[9] pry(main)> cell_2.place_ship(cruiser)
=> #<Ship:0x00007ff7f2903ac0 @health=3, @length=3, @name="Cruiser">
[10] pry(main)> cell_2.render
=> "."
[11] pry(main)> cell_2.render(true)
=> "S"
[12] pry(main)> cell_2.fire_upon
=> 2
[13] pry(main)> cell_2.render
=> "H"
[14] pry(main)> cruiser.sunk?
=> false
[15] pry(main)> cruiser.hit
=> 1
[16] pry(main)> cruiser.hit
=> 0
[17] pry(main)> cruiser.sunk?
=> true
[18] pry(main)> cell_2.render
=> "X"

Comment (priyapower) [reply test/cell_test.rb:80]: Some of our pry info is slightly different than interaction pattern (the interaction pattern returns nothing whereas ours returns something)

INSTANCES:
1. place_ship(cruiser).  -> I can go check ship class
2. fire_upon
3. .hit

I don't know if this due to pry session, do we need to update our code, does it matter, etc...?

**Comment by priyapower** [comment]
Looks awesome!!!

File: test/cell_test.rb (hunk 2 above)
Comment (Arique1104) [reply test/cell_test.rb:80]: Good morning Priya,  I just reviewed all of your comments, but it looks like some of those points were resolved.  So lets focus on this one

I've never checked something like that where our pry sessions shouldn't have return values for these specific methods.  I have typically just been focused on functionality.  

I'm mostly concerned that each method is in fact placing the object or behaving in a way that is predictable.  

I am always willing to challenge my assumptions, but typically if the behavior brought our tests to green, I have moved on.  

I usually use the pry interactions to guide my tests (we don't need to put an assert_equal for `cruiser.hit`, but we do need to put an assert_equal on the method `cell_2.render` returns "X"

Does that make sense?

File: lib/cell.rb (hunk 1 above)
Comment (Arique1104) [reply lib/cell.rb:9]: Don't feel silly...I hate writing instance variables that closely resemble method names, but when the pry interaction requests a `?` that is significant, so this is the work around.  I guess the only fix is to make the instance variable past tense to suggest that it is a boolean, but I think I just made that up and it's not a steadfast truth or norm in programming.

File: test/cell_test.rb (hunk 2 above)
Comment (Arique1104) [reply test/cell_test.rb:80]: [Here's a doc that I found](http://ruby-for-beginners.rubymonstas.org/writing_methods/return_values.html) when I searched for `how to have methods not return values in pry iteraction ruby`. I think we'll be good.

**Pull request merged**